### ISSUE TITLE
[Suggestion] Add things like "structure main toggle" as a on/off button for easier use

### ISSUE THREAD

**Issue by Sar4h-3** [maruohon/minihud#93]
I was struggeling to get the render overlays working, as i had no way to know i first had to activate "structure main toggle" with a hotkey. 
It would be good if you mentioned it somewhere in the "tooltips" or so.
I think it would be easier and more convenient if there was a tab to activate the renderer Options instead of just Hotkeys.
Another option would be to add a possibility to link the render options and the main toggle like you did in Tweakeroo.

**Comment by legoflame** [comment]
thanks that was a big help

**Comment by maruohon** [comment]
Whoops I forgot to respond again it looks like...

So yes the config menus are currently probably the worst thing about my client mods. The config options are divided to categories purely based on how they are organized in the code, and the current malilib version only supports a fixed set of config types in the config menu, unless the mod in question would extend and override large parts of the menu code.

To fix these issues, I finally started working on the long-planned large GUI refactors this past summer. Those are now mostly done (only Litematica hasn't been fully updated to those changes yet), and the new config menus are a lot more flexible in code, and they allow mods to easily define custom config types and the corresponding widget to handle that config in the menu. So basically what this means, is that I can finally easily better group and organize the config menus for the mods. For example MiniHUD currently has three categories just for the Info Line related things, those can be merged into one. And the overlay renderers will finally get their separate toggle options in the menu, so you don't have to use hotkeys to turn everything on anymore... And all the renderer "main" toggle options will be included in all the categories where they are relevant.

![2020-08-19_01 51 01](https://user-images.githubusercontent.com/2331050/102510215-0d726080-4090-11eb-9e63-bdef74760bd6.png)
![2020-08-19_19 25 55](https://user-images.githubusercontent.com/2331050/102509807-a3f25200-408f-11eb-9809-55abc0d1567a.png)
![2020-08-19_19 25 40](https://user-images.githubusercontent.com/2331050/102509820-a6ed4280-408f-11eb-9bf0-b26f02a7d490.png)

**Comment by Awakened-Redstone** [comment]
Oh. That looks awesome!

**Comment by Awakened-Redstone** [comment]
I really want that on tweakeroo too

**Comment by maruohon** [comment]
Yes obviously all the mods will use the new config screens and improved categories.
![2020-09-28_00 22 56](https://user-images.githubusercontent.com/2331050/102920275-1f3e7400-4493-11eb-87f5-23a8667a0504.png)

And also for example the really awkward and messy text-based item and block lists now have proper grouped configs.
![2020-12-22_20 24 16](https://user-images.githubusercontent.com/2331050/102920756-f66aae80-4493-11eb-867a-4907055a4c6d.png)
![2020-12-22_20 24 41](https://user-images.githubusercontent.com/2331050/102920771-fb2f6280-4493-11eb-9424-78c4360896eb.png)


And the item/block/other game object selection is now a dropdown of all the existing objects, instead of just a text field where you input the item's registry name.
![2020-12-22_20 24 48](https://user-images.githubusercontent.com/2331050/102920814-0c786f00-4494-11eb-9faf-2f39fab0c38a.png)
![2020-12-22_20 25 03](https://user-images.githubusercontent.com/2331050/102920825-100bf600-4494-11eb-976d-f6b014c4ea8c.png)
